pkg/reconciler/v1alpha1/pipelinerun/resources/apply_test.go

```diff
@@ -0,0 +1,97 @@
+/*
+ Copyright 2019 Knative Authors LLC
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package resources
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1"
+	tb "github.com/knative/build-pipeline/test/builder"
+)
+
+func TestApplyParameters(t *testing.T) {
+	tests := []struct {
+		name     string
+		original *v1alpha1.Pipeline
+		run      *v1alpha1.PipelineRun
+		dp       []v1alpha1.PipelineParam
```

Comment:
`dp` is part of `original` object. Test could be simplified if params are extracted from original object directly. It ties to my previous comment about simplifying the function signature

Reply:
Good point, will do.